packages/block-library/src/button/style.scss

```diff
@@ -1,8 +1,12 @@
 $blocks-button__height: 56px;
 
-.wp-block-button {
-	color: $white;
+:root .wp-block-button {
+	background: var(--wp--color--gradient);
+	background-color: var(--wp--color--background, $dark-gray-700);
+	color: var(--wp--color--text, $white);
```

Comment:
If both variables --wp--color--gradient and --wp--color--background are set we will always render the gradient background.
That is problematic: imagining we add gradients backgrounds to the group. The group will set --wp--color--gradient as a variable and all buttons inside that group will not be able to choose a custom background color because the variable --wp--color--the background is totally ignored.
To test this we can add `--wp--color--gradient: linear-gradient(135deg,rgba(252,185,0,1) 23%,rgba(255,105,0,1) 100%);` as a style to the body of the page and we can see the custom colors of the buttons are tottaly ignored.